File: CS/CustomDataExplorer/MyDataSourceCollectorService.cs
using DevExpress.Data.Design;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CustomDataExplorer {
    public class MyDataSourceCollectorService : IDataSourceCollectorService {
        private readonly IDataSourceCollectorService innerService;
        public MyDataSourceCollectorService(IDataSourceCollectorService oldService) {
            innerService = oldService;
        }
        public object[] GetDataSources() {
            object[] dataSources = innerService.GetDataSources();
            List<object> result = new List<object>();
            foreach (object item in dataSources) {
                if (!(item is DevExpress.XtraReports.Native.Parameters.ParametersDataSource)) {
                    result.Add(item);
                }
            }
            return result.ToArray();
        }
    }
}

File: VB/CustomDataExplorer/MyDataSourceCollectorService.vb
Imports DevExpress.Data.Design
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace CustomDataExplorer
    Public Class MyDataSourceCollectorService
        Implements IDataSourceCollectorService

        Private ReadOnly innerService As IDataSourceCollectorService
        Public Sub New(ByVal oldService As IDataSourceCollectorService)
            innerService = oldService
        End Sub
        Public Function GetDataSources() As Object() Implements IDataSourceCollectorService.GetDataSources
            Dim dataSources() As Object = innerService.GetDataSources()
            Dim result As New List(Of Object)()
            For Each item As Object In dataSources
                If Not (TypeOf item Is DevExpress.XtraReports.Native.Parameters.ParametersDataSource) Then
                    result.Add(item)
                End If
            Next item
            Return result.ToArray()
        End Function
    End Class
End Namespace
